sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusMetricsProvider.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.ServiceBus.Listeners;
+using Microsoft.Azure.WebJobs.ServiceBus;
+using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Administration;
+using Microsoft.Azure.WebJobs.Extensions.ServiceBus.Config;
+
+namespace Microsoft.Azure.WebJobs.Extensions.ServiceBus.Listeners
+{
+    internal class ServiceBusMetricsProvider
+    {
+        private const string DeadLetterQueuePath = @"/$DeadLetterQueue";
+
+        private readonly ILogger _logger;
+        private readonly string _entityPath;
+        private readonly ServiceBusEntityType _serviceBusEntityType;
+        private readonly Lazy<ServiceBusReceiver> _receiver;
+        private readonly bool _isListeningOnDeadLetterQueue;
+        private readonly Lazy<ServiceBusAdministrationClient> _administrationClient;
+
+        private DateTime _nextWarningTime;
+
+        public ServiceBusMetricsProvider(
+            string entityPath,
+            ServiceBusEntityType serviceBusEntityType,
+            Lazy<ServiceBusReceiver> receiver,
+            Lazy<ServiceBusAdministrationClient> administrationClient,
+            ILoggerFactory loggerFactory)
+        {
+            _serviceBusEntityType = serviceBusEntityType;
+            _receiver = receiver;
+            _entityPath = entityPath;
+            _isListeningOnDeadLetterQueue = entityPath.EndsWith(DeadLetterQueuePath, StringComparison.OrdinalIgnoreCase);
+            _administrationClient = administrationClient;
+            _logger = loggerFactory.CreateLogger<ServiceBusMetricsProvider>();
+            _nextWarningTime = DateTime.UtcNow;
+        }
+
+        public async Task<ServiceBusTriggerMetrics> GetMetricsAsync(bool skipPartitionCount = false)
```

Comment:
Why do we want to skip partition counts? E.g. for TBS just like the existing code for scale monitor, we don't want to scale beyond the number of partitions, right?

Reply:
As discussed offline we can scale beyond the partition count